tests/functional/oauth_reset_password.js

```diff
@@ -49,211 +59,112 @@ define([
-        })
-        .then(function (verificationLink) {
-          return self.remote.get(require.toUrl(verificationLink));
-        })
+        .then(testElementExists('#fxa-confirm-reset-password-header'))
 
-        .then(function () {
-          return FunctionalHelpers.fillOutCompleteResetPassword(
-              self, PASSWORD, PASSWORD);
-        })
+        .then(openVerificationLinkInSameTab(email, 0))
+        .then(fillOutCompleteResetPassword(this, PASSWORD, PASSWORD))
 
-        .findByCssSelector('#loggedin')
-        .end();
+        .then(testElementExists('#loggedin'));
     },
 
     'reset password, verify in a different browser, from the original tab\'s P.O.V.': function () {
-      var self = this;
-
-      return FunctionalHelpers.openFxaFromRp(self, 'signin')
-        .findByCssSelector('a[href="/reset_password"]')
-          .click()
-        .end()
-
-        .findById('fxa-reset-password-header')
-        .end()
+      return this.remote
+        .then(openFxaFromRp(this, 'signin'))
+        .then(click('.reset-password'))
 
+        .then(testElementExists('#fxa-reset-password-header'))
         .then(fillOutResetPassword(email))
 
-        .findById('fxa-confirm-reset-password-header')
-        .end()
-
-        .then(function () {
-          return FunctionalHelpers.openPasswordResetLinkDifferentBrowser(
-                      email, PASSWORD);
-        })
+        .then(testElementExists('#fxa-confirm-reset-password-header'))
+        .then(openPasswordResetLinkDifferentBrowser(email, PASSWORD))
 
         // user verified in a new browser, they have to enter
         // their updated credentials in the original tab.
-        .findByCssSelector('#fxa-signin-header')
-        .end()
-
-        .then(FunctionalHelpers.visibleByQSA('.success'))
-        .end()
-
-        .findByCssSelector('#password')
-          .type(PASSWORD)
-        .end()
-
-        .findByCssSelector('button[type=submit]')
-          .click()
-        .end()
+        .then(testElementExists('#fxa-signin-header'))
+        .then(visibleByQSA('.success'))
+        .then(type('#password', PASSWORD))
+        .then(click('button[type=submit]'))
 
         // user is redirected to RP
-        .findByCssSelector('#loggedin')
-        .end();
+        .then(testElementExists('#loggedin'));
     },
 
     'reset password, verify in a different browser, from the new browser\'s P.O.V.': function () {
-      var self = this;
-
-      return FunctionalHelpers.openFxaFromRp(self, 'signin')
-        .findByCssSelector('a[href="/reset_password"]')
-          .click()
-        .end()
-
-        .findById('fxa-reset-password-header')
-        .end()
+      return this.remote
+        .then(openFxaFromRp(this, 'signin'))
+        .then(click('.reset-password'))
 
+        .then(testElementExists('#fxa-reset-password-header'))
         .then(fillOutResetPassword(email))
 
-        .findById('fxa-confirm-reset-password-header')
```

Comment:
This was the problem right here. We called `findById` without calling `.end()` before calling `clearBrowserState`